### PR TITLE
`azurerm_container_app_environment`: Add support for `dapr_ai_connection_string`

### PR DESCRIPTION
**Pull request by jiaweitao001**
<details>
<summary>
Test results: 
</summary>
<br>

```
--- PASS: TestAccContainerAppEnvironment_basic (670.15s)
--- PASS: TestAccContainerAppEnvironment_requiresImport (731.12s)
--- PASS: TestAccContainerAppEnvironment_complete (1616.27s)
```

</details>

`TestAccContainerAppEnvironment_completeUpdate` fails from time to time due to quota issue, no new failures introduced.

### CODE REVIEW

File: internal/services/containerapps/container_app_environment_resource.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ type ContainerAppEnvironmentModel struct {
 	LogAnalyticsWorkspaceId     string                 `tfschema:"log_analytics_workspace_id"`
 	InfrastructureSubnetId      string                 `tfschema:"infrastructure_subnet_id"`
 	InternalLoadBalancerEnabled bool                   `tfschema:"internal_load_balancer_enabled"`
+	DaprAIConnectionString      string                 `tfschema:"dapr_ai_connection_string"`
```
Comment (jackofallops) [inline internal/services/containerapps/container_app_environment_resource.go:30]: Can we make this long hand for clarity?
```suggestion
	DaprAIConnectionString      string                 `tfschema:"dapr_application_insights_connection_string"`
```

File: internal/services/containerapps/container_app_environment_resource.go [hunk 2]
```diff
@@ -88,6 +89,12 @@ func (r ContainerAppEnvironmentResource) Arguments() map[string]*pluginsdk.Schem
 			Description: "Should the Container Environment operate in Internal Load Balancing Mode? Defaults to `false`. **Note:** can only be set to `true` if `infrastructure_subnet_id` is specified.",
 		},
 
+		"dapr_ai_connection_string": {
```
Comment (jackofallops) [inline internal/services/containerapps/container_app_environment_resource.go:92]:
```suggestion
		"dapr_application_insights_connection_string": {
```

File: internal/services/containerapps/container_app_environment_resource.go [hunk 3]
```diff
@@ -88,6 +89,12 @@ func (r ContainerAppEnvironmentResource) Arguments() map[string]*pluginsdk.Schem
...
+			Type:        pluginsdk.TypeString,
+			Optional:    true,
+			Description: "Application Insights connection string used by Dapr to export Service to Service communication telemetry",
```
Comment (jackofallops) [inline internal/services/containerapps/container_app_environment_resource.go:95]:
```suggestion
			Description: "The Application Insights connection string to be used by Dapr to export Service to Service communication telemetry",
```

File: internal/services/containerapps/container_app_environment_resource.go [hunk 4]
```diff
@@ -191,7 +198,8 @@ func (r ContainerAppEnvironmentResource) Create() sdk.ResourceFunc {
 							SharedKey:  keys.Model.PrimarySharedKey,
 						},
 					},
-					VnetConfiguration: &managedenvironments.VnetConfiguration{},
+					DaprAIConnectionString: pointer.To(containerAppEnvironment.DaprAIConnectionString),
```
Comment (jackofallops) [inline internal/services/containerapps/container_app_environment_resource.go:201]: Can we move this out of the block here and only set when there is a value in config?

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @jiaweitao001 - Thanks for this PR, it's off to a good start.

Can you take a look at the inline comments below, as well as take a look into if the property is updatable or not? If not, it needs marking as `ForceNew` otherwise we should account for it in the Update and ensure there is test coverage for it also.

Thanks!

File: internal/services/containerapps/container_app_environment_resource.go (hunk 1 above)
Comment (jiaweitao001) [reply internal/services/containerapps/container_app_environment_resource.go:30]: Sure. Will do.

File: internal/services/containerapps/container_app_environment_resource.go (hunk 2 above)
Comment (jiaweitao001) [reply internal/services/containerapps/container_app_environment_resource.go:92]: Will do.

File: internal/services/containerapps/container_app_environment_resource.go (hunk 3 above)
Comment (jiaweitao001) [reply internal/services/containerapps/container_app_environment_resource.go:95]: Will fix.

File: internal/services/containerapps/container_app_environment_resource.go (hunk 4 above)
Comment (jiaweitao001) [reply internal/services/containerapps/container_app_environment_resource.go:201]: Sure.

**Comment by jiaweitao001** [comment]
> Hi @jiaweitao001 - Thanks for this PR, it's off to a good start.
> 
> Can you take a look at the inline comments below, as well as take a look into if the property is updatable or not? If not, it needs marking as `ForceNew` otherwise we should account for it in the Update and ensure there is test coverage for it also.
> 
> Thanks!

This property is updatable, I'll add it to the update test.

**Comment by Phiph** [comment]
I wonder if this PR https://github.com/hashicorp/terraform-provider-azurerm/pull/23080 solves this and is merged in

**Comment by stephybun** [comment]
Closing this PR since support for this was merged in #23080. Thanks for pointing that out @Phiph!

**Pull request closed without merging**